x-pack/plugins/timelines/server/search_strategy/timeline/index.ts

```diff
@@ -5,48 +5,167 @@
-      );
+
+      const queryFactory: TimelineFactory<T> = timelineFactory[factoryQueryType];
+
+      if (entityType != null && entityType === EntityType.ALERTS) {
+        return timelineAlertsSearchStrategy({
+          es: esAsInternal,
+          request,
+          options,
+          deps,
+          queryFactory,
+          alerting,
+          alertConsumers: alertConsumers ?? [],
+        });
+      } else {
+        return timelineSearchStrategy({ es, request, options, deps, queryFactory });
+      }
     },
     cancel: async (id, options, deps) => {
       if (es.cancel) {
         return es.cancel(id, options, deps);
+      } else if (esAsInternal.cancel) {
+        return esAsInternal.cancel(id, options, deps);
       }
     },
   };
 };
+
+const timelineSearchStrategy = <T extends TimelineFactoryQueryTypes>({
+  es,
+  request,
+  options,
+  deps,
+  queryFactory,
+}: {
+  es: ISearchStrategy;
+  request: TimelineStrategyRequestType<T>;
+  options: ISearchOptions;
+  deps: SearchStrategyDependencies;
+  queryFactory: TimelineFactory<T>;
+}) => {
+  const { entityType, alertConsumers, ...rest } = request;
+  const dsl = queryFactory.buildDsl(request);
+  return es.search({ request: { ...rest }, params: dsl }, options, deps).pipe(
+    map((response) => {
+      return {
+        ...response,
+        ...{
+          rawResponse: shimHitsTotal(response.rawResponse, options),
+        },
+      };
+    }),
+    mergeMap((esSearchRes) => queryFactory.parse(request, esSearchRes))
+  );
+};
+
+const timelineAlertsSearchStrategy = <T extends TimelineFactoryQueryTypes>({
+  es,
+  request,
+  options,
+  deps,
+  queryFactory,
+  alerting,
+  alertConsumers,
+}: {
+  es: ISearchStrategy;
+  request: TimelineStrategyRequestType<T>;
+  options: ISearchOptions;
+  deps: SearchStrategyDependencies;
+  alerting: AlertPluginStartContract;
+  queryFactory: TimelineFactory<T>;
+  alertConsumers: ALERTS_CONSUMERS[];
+}) => {
+  const allFeatureIdsValid = alertConsumers.every((id) => isValidFeatureId(id));
+
+  if (!allFeatureIdsValid) {
+    throw new Error('An invalid alerts consumer feature id was provided');
+  }
+
+  const indices = alertConsumers.flatMap((consumer) => mapConsumerToIndexName[consumer]);
+  const requestWithAlertsIndices = { ...request, defaultIndex: indices, indexName: indices };
+
+  // Note: Alerts RBAC are built off of the alerting's authorization class, which
+  // is why we are pulling from alerting, not ther alertsClient here
+  const alertingAuthorizationClient = alerting.getAlertingAuthorizationWithRequest(deps.request);
+
+  const getAuthFilter = async () =>
+    alertingAuthorizationClient.getFindAuthorizationFilter(AlertingAuthorizationEntity.Alert, {
+      type: AlertingAuthorizationFilterType.ESDSL,
+      fieldNames: {
+        consumer: OWNER,
+        ruleTypeId: RULE_ID,
+        spaceIds: SPACE_IDS,
+      },
+      includeSpaceId: true,
+    });
```

Comment:
Does it make sense to always include this filter with any consumer of `searchAsInternalUser`?

Reply:
The filter is specific to alerts so I don't think it would make sense to be added to any consumptions of `searchAsInternalUser`, though it is something to consider that perhaps we do want to have some sort of auth layer on top of it? Similar to how adding tags to the route causes security plugin to intercept a request and check if user is authorized in that space.